### ISSUE TITLE
Markdown math latex rendering breaks notebook

### ISSUE THREAD

**Issue by nabriis** [microsoft/vscode#149295]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.67.0 (user setup)
- OS Version: Windows_NT x64 10.0.19042

Steps to Reproduce:

1. Create a .ipynb file and launch it in VS Code
2. Put the following lines in a markdown cell and execute
``` markdown
This code renders fine

\begin{align} \mathbf{A} \end{align}

This code renders as part of the KaTeX formula!
```
3. Result looks like this:
![image](https://user-images.githubusercontent.com/26246495/167948447-f95a5867-3afc-4a47-962e-63b7ff51968e.png)
4. Result looks fine in jupyter notebook.
![image](https://user-images.githubusercontent.com/26246495/167949146-c12a77b3-953c-49b0-afed-5b64880be011.png)
5. If saving the file from 3 with the markdown cell executed in VS Code, it can also no longer be opened by jupyter!

Notes:
I found this error because tools like https://github.com/sphinx-gallery can automatically generate .ipynb from .rst syntax. Here the above markdown cell would be generated from this source:
``` rst
This code renders fine

.. math::

   \mathbf{A}

This code renders as part of the KaTeX formula!

**Comment by wannieman98** [comment]
hi can I take a stab at the issue?

**Comment by mjbvz** [comment]
Yes the relevant code is in https://github.com/mjbvz/markdown-it-katex

**Comment by wannieman98** [comment]
> Yes the relevant code is in https://github.com/mjbvz/markdown-it-katex

Is the relevant code you mentioned important because it affects the rendering here in [extensions/markdown-math/notebook/katex.ts](https://github.com/microsoft/vscode/blob/7e6b007dd579d6a12bb505f7942ad28a5e288b7b/extensions/markdown-math/notebook/katex.ts) ? If yes, then should the change made to the the [https://github.com/mjbvz/markdown-it-katex](https://github.com/mjbvz/markdown-it-katex) repo and make pr there? Thank you

**Comment by wannieman98** [comment]
I think the fundamental problem here is in how the bare math block is handled in bare_math_block function in  https://github.com/mjbvz/markdown-it-katex/blob/2e3736e4b916ee64ed92ebfabeaa94643612665a/index.js. It apparently cannot handle a line with \begin and \end on the same line and it is also related to #149593. 
@mjbvz

**Comment by wannieman98** [comment]
Can I get a tip on how I could debug the markdown-it extension locally? Thanks!!

**Comment by IdoHartuv** [comment]
I don't think there is a issue here.
You need to wrap the line `\begin{align} \mathbf{A} \end{align}` with `$$` like in [their documentation](https://github.com/mjbvz/markdown-it-katex#block).

**Comment by mjbvz** [comment]
Jupyter notebooks also seem to support bare `\begin` equations. Try it on Jupyter labs

(not that I think this design decision is a good one :)

**Comment by legendarykamal** [comment]
hi @mjbvz @nabriis I am new to the community, can I take up this issue and get assigned?

**Comment by TylerLeonhardt** [comment]
![image](https://user-images.githubusercontent.com/2644648/214712597-b04aa256-2606-424f-8824-cb3a74b93b62.png)

Looks good
